src/System.Buffers/src/System/Buffers/DefaultArrayPoolBucket.cs

```diff
@@ -80,7 +98,17 @@ internal void Return(T[] buffer)
                 // we can just drop this buffer
                 if (_index != 0)
                 {
+                    // If we have a debugger attached, do an extra safety check
+                    // to prevent a double-return scenario
+                    if (System.Diagnostics.Debugger.IsAttached)
```

Comment:
IsAttached looks to be relatively cheap...it involves a call down to Native code but that code just checks for a flag. It'd be nice if we could listen for an event saying a debugger has been attached instead of checking each time, but oh well